### PR TITLE
LPS-190225 Discard deleted fragment entry links when load/reload a page

### PR DESCRIPTION
**Pull request by YurenaC**
[Story](https://liferay.atlassian.net/browse/LPS-189636)

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

**Comment by YurenaC** [comment]
ci:test:sf

**Comment by YurenaC** [comment]
ci:test:relevant

File: modules/apps/layout/layout-content-page-editor-web/src/main/java/com/liferay/layout/content/page/editor/web/internal/display/context/ContentPageEditorDisplayContext.java [hunk 1]
```diff
@@ -1354,9 +1370,11 @@ private Map<String, Object> _getFragmentEntryLinksMap(
 				}
 			}
 
-			fragmentEntryLinksMap.put(
-				String.valueOf(fragmentEntryLink.getFragmentEntryLinkId()),
-				jsonObject);
+			if (!fragmentEntryLink.isDeleted()) {
```
Comment (ealonso) [inline modules/apps/layout/layout-content-page-editor-web/src/main/java/com/liferay/layout/content/page/editor/web/internal/display/context/ContentPageEditorDisplayContext.java:1373]: @YurenaC the idea behind this task is avoid unnecessary method calls, if we add it to the end we are not avoiding this calculations

File: modules/apps/layout/layout-content-page-editor-web/src/main/java/com/liferay/layout/content/page/editor/web/internal/display/context/ContentPageEditorDisplayContext.java [hunk 2]
```diff
@@ -1264,15 +1265,29 @@ private Map<String, Object> _getFragmentEntryLinks() throws Exception {
...
 			layoutStructure.getFragmentLayoutStructureItems();
 
-		for (long fragmentEntryLinkId : fragmentLayoutStructureItems.keySet()) {
+		Set<Map.Entry<Long, LayoutStructureItem>> entries =
+			fragmentLayoutStructureItems.entrySet();
+
+		Iterator<Map.Entry<Long, LayoutStructureItem>> iterator =
+			entries.iterator();
+
+		while (iterator.hasNext()) {
```
Comment (ealonso) [inline modules/apps/layout/layout-content-page-editor-web/src/main/java/com/liferay/layout/content/page/editor/web/internal/display/context/ContentPageEditorDisplayContext.java:1274]: @YurenaC Is it not possible to return only not deleted fragments in service calls?

**Pull request closed without merging**